# ReadMe

Loads all the sources listed under China from the Factiva website (accessed Aug 10, 2026), and filters down to mainland china media sources to see how many there are.

In [22]:
import pandas as pd
import os

from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [23]:
with open("../data/factiva_china_sources.csv", "r") as raw_text:
    treated_text = raw_text.read().splitlines()  # reads each line as one string

df = pd.DataFrame(treated_text, columns=["media"])

In [29]:
df.shape[0]
df.head()
df.iloc[0]

665

,media
0,"10000 Link (China, Chinese Language - Simplif..."
1,"Web Site:100EC.CN (China, Chinese Language - ..."
2,"Web Site:100PPI (China, Chinese Language - Si..."
3,Web Site:163.com (Chinese Language - Simplifi...
4,Web Site:21IC China Electronics Network (Chin...


media     10000 Link (China, Chinese Language - Simplif...
Name: 0, dtype: object

In [30]:
# filters to sources only in Simplified Chinese, meaning it's from mainland China.
df_filtered = df[ df.media.str.contains("Simplified") ]

In [33]:
df_filtered.shape[0]
df_filtered.sample(10)

# check if it contains People's Daily (one of the most importnat newspapers)
df_filtered[ df_filtered.media.str.contains("Daily") ]

425

,media
186,Web Site:China Shipping Information (Chinese ...
148,Web Site:China News Service Guangdong News (C...
656,"Publication:YiMagazine (China, Chinese Langua..."
154,Web Site:China News Service Jilin News (Chine...
299,Web Site:Guandong Provincial Public Security ...
131,Web Site:China LED Net (Chinese Language - Si...
590,Publication:TravelDaily China (Chinese Langua...
569,"Publication:TechNode (China, Chinese Language..."
492,Publication:PR Newswire Asia-OS (Chinese Lang...
36,Web Site:Beijing Discipline Inspection and Su...


,media
31,Web Site:Ban Dao Daily (Chinese Language - Si...
35,Web Site:Beijing Daily (Chinese Language - Si...
73,Publication:Chengdu Economic Daily-Hongxing N...
94,Web Site:China Daily (Chinese Language – Simp...
217,Publication:Chutian Metropolis Daily Online -...
258,Web Site:Economic Information Daily (Chinese ...
286,Web Site:Gansu Daily (Chinese Language - Simp...
333,Publication:Hong Kong Shipping Gazette Daily ...
373,"Publication:Jing Daily (China, Chinese Langua..."
388,Publication:Macao Daily News (Chinese Languag...


# Conclusion
Factiva offers 425 different sources on chinese mainland media.